File: 45주차/타겟넘버.py
answer = 0
def solution(numbers, target):
    n = len(numbers)
    def dfs(L,total):
        global answer
        if L == n:
            if total == target:
                answer += 1
            return
        dfs(L+1, total+numbers[L])
        dfs(L+1,total-numbers[L])
    dfs(0,0)
 
   
    return answer
# 프로그래머스 dfs 문제

File: 45주차/타겟넘버.js
function solution(numbers, target) {
  var answer = 0;
  const n = numbers.length;
  function dfs(L, value) {
    if (L === n) {
      if (value === target) answer += 1;
      return;
    }
    dfs(L + 1, value + numbers[L]);
    dfs(L + 1, value - numbers[L]);
  }
  dfs(0, 0);

  return answer;
}
//프로그래머스 dfs 문제
